dev/com.ibm.ws.security/src/com/ibm/ws/security/context/SubjectRegistryManager.java

```diff
@@ -0,0 +1,113 @@
+ * off the thread and provides the ability to clear the subject registry info off the thread.
+ * See {@link SubjectRegistryThreadContext} for more information.
+ */
+public class SubjectRegistryManager {
+
+    private static ThreadLocal<SubjectRegistryThreadContext> threadLocal = new SecurityThreadLocal();
+
+    /**
+     * Gets the subject registry thread context that is unique per thread.
+     * If/when a common thread storage framework is supplied, then this method
+     * implementation may need to be updated to take it into consideration.
+     *
+     * @return the subject registry thread context.
+     */
+    @Trivial
+    protected static SubjectRegistryThreadContext getSubjectRegistryThreadContext() {
+        ThreadLocal<SubjectRegistryThreadContext> currentThreadLocal = getThreadLocal();
+        SubjectRegistryThreadContext subjectRegistryThreadContext = currentThreadLocal.get();
+        if (subjectRegistryThreadContext == null) {
+            subjectRegistryThreadContext = new SubjectRegistryThreadContext();
+            currentThreadLocal.set(subjectRegistryThreadContext);
+        }
+        return subjectRegistryThreadContext;
+    }
+
+    /**
+     * Set whether or not the subject is from the SAF registry.
+     *
+     * @param isSAF True if the subject is from the SAF registry.
+     */
+    public static void setSubjectIsSAF(boolean isSAF) {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.setIsSAF(isSAF);
+    }
+
+    /**
+     * Start subject registry detection. This is only required if a SAF
+     * registry is configured.
+     *
+     * @param isSAFRegistryConfigured
+     */
+    public static void startSubjectRegistryDetectionOnZOS() {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.detect();
+    }
+
+    /**
+     * Clear the subject registry detection. This is only required if a SAF
+     * registry is configured.
+     *
+     * @param isSAFRegistryConfigured
+     */
+    public static void clearSubjectRegistryDetectionOnZOS() {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.donotdetect();
+    }
+
+    /**
+     * Determine if a SAF credential should be created.
+     * The instances when a SAF credential should be
+     * created are:
+     * <ol>
+     * <li>SAF registry is configured and subject is from the SAF registry</li>
+     * <li>mapDistributedIdentities is true</li>
+     * <li>Subject is from the OS registry (and not logging in)</li>
+     * </ol>
+     *
+     * @return True if a SAF credential should be created and false if it should not
+     */
+    public static boolean isCreateSAFCredential() {
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
```

Comment:
Probably do the `Platform.isZOS()` check here.

```suggestion
        if (!Platform.isZOS()) {
            return false;
        }
        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
```

Reply:
(this looks like it is only called from SAFCredentialsServiceImpl... maybe don't technically need to check)